nautobot/extras/management/commands/runjob.py

```diff
@@ -74,23 +74,26 @@ def handle(self, *args, **options):
             job_result = JobResult.objects.get(pk=job_result.pk)
 
         # Report on success/failure
-        for test_name, attrs in job_result.data.items():
-
-            if test_name in ["total", "output"]:
-                continue
+        groups = set(JobLogEntry.objects.filter(job_result=job_result).values_list("grouping", flat=True))
+        for group in sorted(groups):
+            logs = JobLogEntry.objects.filter(job_result__pk=job_result.pk, grouping=group)
+            success_count = logs.filter(log_level=LogLevelChoices.LOG_SUCCESS).count()
+            info_count = logs.filter(log_level=LogLevelChoices.LOG_INFO).count()
+            warning_count = logs.filter(log_level=LogLevelChoices.LOG_WARNING).count()
+            failure_count = logs.filter(log_level=LogLevelChoices.LOG_FAILURE).count()
 
             self.stdout.write(
                 "\t{}: {} success, {} info, {} warning, {} failure".format(
-                    test_name,
-                    attrs["success"],
-                    attrs["info"],
-                    attrs["warning"],
-                    attrs["failure"],
+                    group,
+                    success_count,
+                    info_count,
+                    warning_count,
+                    failure_count,
                 )
             )
 
-            for log_entry in attrs["log"]:
-                status = log_entry[1]
+            for log_entry in logs:
```

Comment:
Consider using `logs.iterator()` here since these are single-use instances that you're not mutating.

Reply:
See: https://docs.djangoproject.com/en/3.2/ref/models/querysets/#iterator

Reply:
I will look into iterator, but this is only used when running jobs through management commands, which has limited use as they can't take arguments.